### PR TITLE
Take advantage of RStudio 1.3 preference handling

### PR DESCRIPTION
**Pull request by hadley**
* Look for snippets in new location
* New helper to see/edit prefs file
* `use_blank_slate()` now automated

### CODE REVIEW

File: R/edit.R [hunk 1]
```diff
@@ -100,13 +101,26 @@ edit_r_makevars <- function(scope = c("user", "project")) {
...
+
+  # Snippet location changed in 1.3:
+  # https://blog.rstudio.com/2020/02/18/rstudio-1-3-preview-configuration/
+  new_rstudio <- !rstudioapi::isAvailable() || rstudioapi::getVersion() >= "1.3.0"
+  old_path <- path_home_r(".R", "snippets", file)
+  new_path <- rstudio_config_path("snippets", file)
+
+  # Mimic RStudio behaviour: copy to new location if you edit
+  if (new_rstudio && file_exists(old_path) && !file_exists(new_path)) {
+    dir_create(path_dir(new_path))
```
Comment (jennybc) [inline R/edit.R:118]: Are you intentionally not using our helper `create_directory()`? I don't have a strong opinion but weakly think we should use it or have a reason not to.

File: R/rstudio.R [hunk 2]
```diff
@@ -39,33 +39,17 @@ use_rstudio <- function(line_ending = c("posix", "windows")) {
...
-      "In {ui_field('Global Options > General')}, \\
-      set {ui_field('Save workspace to .RData on exit')} to {ui_value('Never')}."
-    )
-    ui_todo(
-      "Call {ui_code('use_blank_slate(\"project\")')} to opt in to the \\
-      blank slate workflow for a specific project."
-    )
+  if (scope == "user") {
+    use_rstudio_config(list(
+      save_workspace = "never",
```
Comment (jennybc) [inline R/rstudio.R:50]: Do you think we should announce anything about what we're doing? I guess it's a simple function and this is the only reason to call it.

File: R/rstudio.R [hunk 3]
```diff
@@ -176,3 +160,43 @@ rstudio_git_tickle <- function() {
...
+  path <- rstudio_config_path("rstudio-prefs.json")
+  if (file_exists(path)) {
+    jsonlite::read_json(path)
+  } else {
+    list()
+  }
+}
+rstudio_prefs_write <- function(json) {
+  path <- rstudio_config_path("rstudio-prefs.json")
+  dir_create(path_dir(path))
```
Comment (jennybc) [inline R/rstudio.R:182]: Ditto re: `create_directory()`

**Review by jennybc — COMMENTED** [review]
Basically LGTM. I pushed a few tweaks and posed a few questions.

File: R/rstudio.R (hunk 2 above)
Comment (hadley) [reply R/rstudio.R:50]: `use_rstudio_config()` should do the notification

File: R/edit.R (hunk 1 above)
Comment (hadley) [reply R/edit.R:118]: No, just because I forgot about it 😄

File: NEWS.md [hunk 4]
```diff
@@ -131,6 +131,16 @@ The default Git protocol is now "https" and we no longer provide an interactive
   been removed from all other license functions because it is not needed;
   no other license makes an assertion about who the copyright holder is.
 
+## RStudio preferences
+
+usethis is now fully cognizant of the [changes to RStudio preferences](https://blog.rstudio.com/2020/02/18/rstudio-1-3-preview-configuration/) in RStudio 1.3:
+
+* `edit_rstudio_snippets()` looks in the new location, and if you have snippets in the old location, will automatically copy them to the new location (#1204)
```
[the remark this answers is not in the document]
Comment (hadley) [reply NEWS.md:138]: In general, I think moving to the new location would be a better strategy, but consistency with the IDE is the dominant concern here.

**Pull request merged**